webidl.txt

```diff
@@ -148,51 +176,74 @@ dictionary VideoEncodeLayer {
   unsigned long? bitsPerSecond;
 }
 
-[Constructor(BufferSource data, MediaTime timestamp)]
-interface EncodedVideoFrame {
-  readonly attribute Uint8Array data;
-  readonly attribute MediaTime timestamp;
-  // Info provided as a result from the encoder
-  // Not needed as input to a decoder
-  readonly attribute VideoEncodeResult? encoded;
+dictionary VideoEncoderStaticSettings {
+  // Can be used to initialize the encoder faster
+  // than waiting for the first frame
+  unsigned long? expectedWidth;
+  unsigned long? expectedHeight;  
+
+  VideoEncodeContentMode contentMode;
+
+  sequence<VideoEncodeLayer> layers;
```

Comment:
Shouldn't this be dynamic to allow group call clients to change which layers they send with depending on the number of people in the room?

Reply:
Perhaps a distinction between static and dynamic parameters is the requirement to reinitialize the codec. Dynamic settings should be something that does not require reinitialization (concretely, does not require a new keyframe). In this case the right approach is to reinitialize the codec.

Reply:
Ah, that's a good idea.

Reply:
Can you put a comment to that effect somewhere in the doc?

Reply:
Yes, it was added to the explainer in the last diff.